### PR TITLE
add goreleaser

### PR DESCRIPTION
**Pull request by cpanato**
**What type of PR is this?**

 /kind feature

**Any specific area of the project related to this PR?**

 /area build

**What this PR does / why we need it**:

Add goreleaser for falcosidekick

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Review by leogr — CHANGES_REQUESTED** [review]
The previous job for `build-push-main` should be removed since goreleaser will take care of everything, shouldn't?

**Comment by cpanato** [comment]
@leodido the goreleaser will take care only for tags, the `build-push-main` will just build/push the latest tag everytime we push something to the main branch

so I guess it is ok to keep

**Comment by leogr** [comment]
> @leogr the goreleaser will take care only for tags, the `build-push-main` will just build/push the latest tag everytime we push something to the main branch

A tag will be on the master, and both `build-push-main` and goreleaser will push the latest tag:
https://github.com/falcosecurity/falcosidekick/pull/209/files#diff-42e26dc67aed8aa3edb2472b4403288c1699fb6dc47419b9a475f0f224fe4689R21

I personally would either:
- completely remove `build-push-main` 
- or (if we really need a docker image for the master) make `build-push-main` push to `falcosecurity/falcosidekick:master`

Anyway, I would not use `latest` for unreleased stuff, since many folks expect that `latest` is just pointing to the latest stable version.

File: .circleci/config.yml [hunk 1]
```diff
@@ -60,3 +71,14 @@ workflows:
...
+          context: falco
+          requires:
+            - test
+            - lint
+            - build-image
+          filters:
+            branches:
+              ignore: /.*/
+            tags:
+              only: /v[0-9]+(\.[0-9]+)*(-.*)*/
```
Comment (Issif) [inline .circleci/config.yml:84]: Until now, I never prefixed release version with `v`, do we force the usage or add `v` as optionnal in regexp?

**Comment by Issif** [comment]
> > @leogr the goreleaser will take care only for tags, the `build-push-main` will just build/push the latest tag everytime we push something to the main branch
> 
> A tag will be on the master, and both `build-push-main` and goreleaser will push the latest tag:
> [#209 (files)](https://github.com/falcosecurity/falcosidekick/pull/209/files#diff-42e26dc67aed8aa3edb2472b4403288c1699fb6dc47419b9a475f0f224fe4689R21)
> 
> I personally would either:
> 
>     * completely remove `build-push-main`
> 
>     * or (if we really need a docker image for the master) make `build-push-main` push to `falcosecurity/falcosidekick:master`
> 
> 
> Anyway, I would not use `latest` for unreleased stuff, since many folks expect that `latest` is just pointing to the latest stable version.

What about:
- tag: tagged commits (obvious)
- latest: last commit in master
- stable: last tagged commit

?

File: .circleci/config.yml (hunk 1 above)
Comment (cpanato) [reply .circleci/config.yml:84]: usually is good to prefix, but if we are not using that I will remove

Comment (cpanato) [reply .circleci/config.yml]: ptal

**Review by leogr — APPROVED** [review]

**Pull request merged**